### PR TITLE
Problem: not exposing c++ api for wc 2.0 sign , send tx (fix #453)

### LINKED ISSUE #453: Problem: not exposing c++ api for wc 2.0 sign , send tx
expose c++ api for wc 2.0 sign, send

### PR DESCRIPTION
**Pull request by leejw51crypto**
expose c++ api for wc 2.0 sign, send tx

### CODE REVIEW

**Comment by leejw51crypto** [comment]
ready for cr
fixing ci errors

**Comment by leejw51crypto** [comment]
ci test fails, fixing now

**Comment by leejw51crypto** [comment]
rust linting rule changed, inspecting now

File: wallet-connect/src/v2/client.rs [hunk 1]
```diff
@@ -219,6 +222,22 @@ impl<M: Middleware> MiddlewareError for WCError<M> {
     }
 }
 
+fn pad_zero(s: String) -> String {
```
Comment (rarcifa) [inline wallet-connect/src/v2/client.rs:227]: We could add more documentation to the functions, and describe the params.

File: wallet-connect/src/v2/client.rs [hunk 2]
```diff
@@ -219,6 +222,22 @@ impl<M: Middleware> MiddlewareError for WCError<M> {
...
+    if s.len() % 2 != 0 {
+        format!("0{s}")
+    } else {
+        s
+    }
+}
+
+fn append_hex(s: String) -> String {
```
Comment (rarcifa) [inline wallet-connect/src/v2/client.rs:237]: here too

File: wallet-connect/examples/web3_v2.rs [hunk 3]
```diff
@@ -42,8 +74,132 @@ async fn save(info: &SessionInfo) -> eyre::Result<()> {
     Ok(())
 }
 
+async fn sign_typed_tx(
```
Comment (rarcifa) [inline wallet-connect/examples/web3_v2.rs:77]: documentation maybe on the functions?

**Review by rarcifa — APPROVED** [review]
not familiar with the code, but lgtm overall. Maybe we could add some documentation on a few functions in the future.

File: wallet-connect/examples/web3_v2.rs (hunk 3 above)
Comment (leejw51crypto) [reply wallet-connect/examples/web3_v2.rs:77]: ok

File: wallet-connect/src/v2/client.rs (hunk 1 above)
Comment (leejw51crypto) [reply wallet-connect/src/v2/client.rs:227]: ok

File: wallet-connect/src/v2/client.rs (hunk 2 above)
Comment (leejw51crypto) [reply wallet-connect/src/v2/client.rs:237]: ok

**Comment by leejw51crypto** [comment]
working on comments

**Pull request merged**